react_excercise/got-app/src/Api.js

```diff
@@ -0,0 +1,80 @@
+import axios from 'axios';
+import levenshteinDistance from './LevenshteinDistance';
+
+const url = 'https://thronesapi.com/api/v2/Characters';
+
+async function fetchData() {
+    try {
+        const response = await axios.get(url);
+        return response.data;
+    } catch (error) {
+        console.error('Error fetching data from API');
+        throw error;
+    }
+}
+
+async function searchForCharacters(input) {
+    try {
+        const results = await fetchData();
+        if (!results || results.length === 0) {
+            throw new Error('No data found');
+        }
+        const findCharacter = results.filter((character) => {
+            return character.firstName.toLowerCase() === input.toLowerCase();
+        });
+        if (findCharacter.length === 0) {
+            throw new Error('No characters found');
+        }
+        return findCharacter;
+    } catch (error) {
+        console.error('Error finding character');
+        throw error;
+    }
+}
+
+async function getHouseMembers() {
+    try {
+        const characters = await fetchData();
+        const houseMembers = {};
+
+        // Count characters in each house
+        characters.forEach((character) => {
+            const houseName = character.family;
+            let nameOfHouse;
+            const threshold = 2; // Limit on how many letters could be misspelled, when comparing family names
```

Comment:
Minor - Try to keep constants in separate files and then use it by importing. 
Again, this is small application so it doesn't matter. But when you scale and needs to change some constant, you can do so by just changing values at one place.

Reply:
I'll keep that in mind.  Thanks for the advice